algebird-core/src/main/scala/com/twitter/algebird/HyperLogLog.scala

```diff
@@ -451,3 +451,15 @@ case class HyperLogLogAggregator(val hllMonoid: HyperLogLogMonoid) extends Monoi
   def prepare(value: Array[Byte]) = monoid.create(value)
   def present(hll: HLL) = hll
 }
+
+case class SetSizeAggregator[A](hllBits: Int, maxSetSize: Int = 10)(implicit toBytes: A => Array[Byte])
```

Comment:
it might make sense to make the default maxSetSize to be something that depends on hllBits. At the extreme you could serialize every item coming in and increment a byte counter (although the in-memory may be much less or more, but still), then when it exceeds `2^hllBits`, switch over.  I would think something like `maxSetSize: Int = math.max(10, 1 << (hllBits - 5))` might be a decent default.